src/rtp/header.rs

```diff
@@ -253,50 +253,61 @@ impl RtpHeader {
     /// The logic detects wrap-arounds of the 16-bit RTP sequence number.
     #[doc(hidden)]
     pub fn sequence_number(&self, previous: Option<SeqNo>) -> SeqNo {
-        let e_seq = extend_seq(previous.map(|v| *v), self.sequence_number);
+        let e_seq = extend_u16(previous.map(|v| *v), self.sequence_number);
         e_seq.into()
     }
 }
 
-/// "extend" a 16 bit sequence number into a 64 bit by
-/// using the knowledge of the previous such sequence number.
-pub fn extend_seq(prev_ext_seq: Option<u64>, seq: u16) -> u64 {
-    // We define the index of the SRTP packet corresponding to a given
-    // ROC and RTP sequence number to be the 48-bit quantity
-    //       i = 2^16 * ROC + SEQ.
-    //
-    // https://tools.ietf.org/html/rfc3711#appendix-A
-    //
-    let seq = seq as u64;
-
-    if prev_ext_seq.is_none() {
-        // No wrap-around so far.
-        return seq;
-    }
+macro_rules! mk_extend {
+    ($id:ident, $t:ty, $bits:expr) => {
+        /// "extend" a less than 64 bit sequence number into a 64 bit by
+        /// using the knowledge of the previous such sequence number.
+        pub fn $id(prev_ext_seq: Option<u64>, seq: $t) -> u64 {
+            const MAX: u64 = 2_u64.pow($bits); // u16: 65_536;
```

Comment:
Wouldn't `$ty::MAX` work here? That way the `$bits` argument can be removed

Reply:
Good point. Let's see if I can fix.